### PR TITLE
fix(ui): Fix Select2Field for new settings, fix default role field

### PR DESCRIPTION
**Pull request by billyvg**

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/settings/components/forms/fieldFromConfig.jsx [hunk 1]
```diff
@@ -83,14 +64,15 @@ export default class FieldFromConfig extends React.Component {
         return <TextareaField {...props} />;
       case 'choice':
       case 'select':
+      case 'array':
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/components/forms/fieldFromConfig.jsx]: is there a reason for needing 3 aliases for this?

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx [hunk 2]
```diff
@@ -215,6 +219,7 @@ class FormField extends React.Component {
     } = this.props;
     let id = this.getId();
     let model = this.getModel();
+    let isDisabled = typeof disabled === 'function' ? disabled(this.props) : disabled;
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: 👍

File: src/sentry/static/sentry/app/views/settings/components/forms/jsonForm.jsx [hunk 3]
```diff
@@ -64,6 +65,7 @@ class JsonForm extends React.Component {
               <PanelBody>
                 {fields.map(field => (
                   <FieldFromConfig
+                    access={access}
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/components/forms/jsonForm.jsx]: What does this do? It doesn't appear to be used in `FieldFromConfig`

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx [hunk 4]
```diff
@@ -143,7 +144,10 @@ class FormField extends React.Component {
    * Set field's hover state and propagate callbacks
    */
   handleHover = (mouseOver, ...args) => {
-    let {name, onMouseOver, onMouseOut} = this.props;
+    let {name, onMouseOver, onMouseOut, noHover} = this.props;
+
+    if (noHover) return;
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: Could `noHover` be default? Or do many of the fields have hover states?

File: src/sentry/static/sentry/app/views/settings/components/forms/jsonForm.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/forms/jsonForm.jsx]: access is a Set of permissions for user, it's used in a callback (currently only for `disabled`).

File: src/sentry/static/sentry/app/views/settings/components/forms/fieldFromConfig.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/forms/fieldFromConfig.jsx]: Potential backwards compatibility.

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx (hunk 4 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: I completely removed this, instead removed handling of hover state from react state and moved it into our styled component

**Review by lynnagara — APPROVED** [review]

**Pull request merged**